ietf/draft-ietf-jmap-calendars-18.txt

```diff
@@ -0,0 +1,3360 @@
+   calendar "myRights" allows them to do so.  All other properties may
+   be overwritten when a future update arrives to this event from the
+   origin (e.g., via an iTIP REQUEST message).  Such updates may be
+   directly applied by the server, or applied at the user's request by a
+   client if it has access to the data through some other means (e.g.,
+   the client also has access to the user's email and can parse an iMIP
+   message).
+
+   When updating an event, if all of:
+
+   *  a property has been changed other than "calendarIds", "isDraft",
+      "updated" or a per-user property (see Section 5.3); and
+   *  the server is the origin of the event (the "isOrigin" property is
+      true); and
+   *  the "sequence" property is not explicitly set in the update, or
+      the given value is less than or equal to the current "sequence"
+      value on the server;
+
+   then the server MUST increment the "sequence" value by one.
+
+   The "method" property MUST NOT be set.  Any attempt to do so is
+   rejected with a standard invalidProperties SetError.
+
+   If "utcStart" is set, this is translated into a "start" property
+   using the server's current time zone information.  It MUST NOT be set
+   in addition to a "start" property and it cannot be set inside
+   "recurrenceOverrides"; this MUST be rejected with an
+   invalidProperties SetError.
+
+   Similarly, the "utcEnd" property is translated into a "duration"
+   property if set.  It MUST NOT be set in addition to a "duration"
+   property and it cannot be set inside "recurrenceOverrides"; this MUST
+   be rejected with an invalidProperties SetError.
+
+
+
+
+Jenkins & Douglass       Expires 9 December 2024               [Page 28]
+
+Internet-Draft               JMAP Calendars                    June 2024
+
+
+   The server does not automatically reset the "partipationStatus" or
+   "expectReply" properties of a Participant when changing other event
+   details.  Clients should either be intelligent about whether the
+   change invalidates previous RSVPs, or ask the user whether to reset
+   them.
+
+   The server MAY enforce that all events have an owner, for example in
+   team calendars.  If the user tries to create an event without
+   participants in such a calendar, the server MUST automatically add a
+   participant with the "owner" role corresponding to one of the user's
+   ParticipantIdentities (see Section 3).
+
+   When creating an event with participants, or adding participants to
+   an event that previously did not have participants, the server MUST
+   set the "replyTo" property of the event if not present.  Clients
+   SHOULD NOT set the "replyTo" property for events when the user adds
+   participants; the server is better positioned to add all the methods
+   it supports to receive replies.
+
+   The following extra SetError type is defined:
+
+   *  *noSupportedScheduleMethods*: The server was requested to send
```

Comment:
It's the former — you never have partial application of an individual create/update/delete in JMAP, it either succeeds or fails. This is clarified more in the description of the `sendSchedulingMessages` argument:

_The server may send the scheduling message via any of the methods defined on the sendTo property of a participant (if the server is the origin) or the replyTo property of the event (otherwise) that it supports. If no supported methods are available for any of the recipients, the server MUST reject the change with a <tt>noSupportedScheduleMethods</tt> SetError._

(If the change is rejected, clearly no messages should be sent out.)